subsys/bluetooth/audio/tbs_client.c

```diff
@@ -730,29 +787,16 @@ static uint8_t read_uri_list_cb(struct bt_conn *conn, uint8_t err,
 				    struct bt_gatt_read_params *params,
 				    const void *data, uint16_t length)
 {
-	struct bt_tbs_instance *inst = CONTAINER_OF(params, struct bt_tbs_instance, read_params);
-	uint8_t inst_index = tbs_index(conn, inst);
-	const char *uri_scheme_list = NULL;
+	bt_tbs_client_read_string_cb cb = NULL;
 
-	(void)memset(params, 0, sizeof(*params));
-
-	LOG_DBG("Index %u", inst_index);
-
-	if (err != 0) {
-		LOG_DBG("err: 0x%02X", err);
-	} else if (data != NULL) {
-		uri_scheme_list = parse_string_value(data, length,
-					MAX_URI_SCHEME_LIST_SIZE);
-		LOG_DBG("%s", uri_scheme_list);
-	}
-
-	inst->busy = false;
+	LOG_DBG("Read bearer UCI");
 
 	if (tbs_client_cbs != NULL && tbs_client_cbs->uri_list != NULL) {
-		tbs_client_cbs->uri_list(conn, err, inst_index, uri_scheme_list);
+		cb = tbs_client_cbs->uri_list;
 	}
 
-	return BT_GATT_ITER_STOP;
+	return handle_string_long_read(conn, err, params, data,
+				       length, cb, true);
```

Comment:
The URI list probably should only be truncated if we can truncate it "properly". 

e.g. if the buffer (for the sake of simplicity) is 15 octets (14 chars + 1 null terminator) then `tel,skype,teamspeak` should (if we decide to allow truncating) become `tel,skype` rather than `tel,skype,team` IMO

Reply:
I get your point, "team" is not intended one of the assigned numbers.
I will change the truncatable argument to false.
I guess, that in general it would be nicer to either
1) indicate to the callback if the string has been truncated
2) always fail if we would have to truncate
The application would have to decide the next step in both cases

Reply:
The simplest solution is to just not truncate this :)